specs/tree-view/tree-view.md

```diff
@@ -0,0 +1,195 @@
+
+### Use Cases
+
+Primarily used to present a hierarchical list such as a file system. A tree view can also be served as a flat list where the items will function as navigation.
+  
+### Features
+
+- **Selection:** A tree view can support either single select or multi-select behavior for nodes. **I would propose that for the initial spec we start with supporting single select only.*
+
+
+- **Drag & Drop support:** Often, as is in the case of a "file system" example, drag and drop support is desired. This does present [challenges](#drag-&-drop).
+
+### Risks and Challenges
+
+#### Nesting:
+
+Infinite nesting combined with the current visual design presents a styling challenge. Consider a scenario where we have the following tree in pseudo code:
+```html
+<fast-tree-view>
+    <fast-tree-item>Root node one</fast-tree-item>
+    <fast-tree-item>Root node two</fast-tree-item>
+    <fast-tree-item>
+        Parent Node && Root node
+        <fast-tree-item>Nested node one</fast-tree-item>
+        <fast-tree-item>Nested node two</fast-tree-item>
+    </fast-tree-item>
+    <fast-tree-item>Root node three</fast-tree-item>
+</fast-tree-view>
+```
+In this scenario, there is no root node with an expand/collapse glyph. The third item in the tree view is a parent node, which *would* have an expand/collapse glyph to the right. To ensure that the content of the parent node align with the rest of the root nodes, the entire tree will need to visually shift to ensure alignmnet of the tree content. This presents an interesting challenge for how we will communicate these changes to the parent to ensure the styling is appropriately applied. Ultimately, the parent just needs to know that *one* of the items below has child nodes.
+
+#### Drag & Drop
+
+Drag & Drop support poses a unique set of challenges, especially if our goal is to provide out of the box support via the [HTML Drag and Drop API](https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API). Alternatively, those using our components with React, Vue, Angular, etc may want to use libraries built to work with their specific framework flavor. 
+
+**Considering the challenge that drag & drop would present for any given component, I think it's worth holding on that feature, and taking it up seperately.*
+
+### Prior Art/Examples
+- [FAST-DNA (React)](https://explore.fast.design/components/tree-view)
+- [Material UI](https://material-ui.com/components/tree-view/)
+- [Lightning Design](https://www.lightningdesignsystem.com/components/trees/)
+- [Ant Design](https://ant.design/components/tree/)
+- [Atlassian](https://atlaskit.atlassian.com/packages/core/tree/example/static-tree)
+- [Windows (UWP)](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/tree-view)
+
+---
+
+## Design
+
+*Describe the design of the component, thinking through several perspectives:*
+
+- *A customer using the component on a web page.*
```

Comment:
Feel free to remove boilerplate...

Reply:
Thanks - I think we could add this to the top of the spec; I got a couple comments on here, but I was torn on a couple ways to go:
1. Leave boilerplate so people can know what's been ommitted
2. Omit the boilerplate as it's not used.

If our hope is the second, it might help to have that called out explicitly.